mgr/dashboard: Cleaner notifications

Devp00l opened the pull request:
Now background tasks and recent notifications won't differ in their
wording anymore and all notifications have the same style.

Resolves issue #24460

Signed-off-by: Stephan Müller <smueller@suse.com>

tspmelo commented:
You should write the url for the issue in the commit msg:
`Fixes: https://tracker.ceph.com/issues/24460`

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts:
```diff
@@ -6,180 +6,136 @@ import { Task } from '../models/task';
-    this.success = success;
-    this.error = error;
+  errors: (metadata) => object;
+  prefixes: string[];
```
Would it make sense if we had properties for each of the prefixes, instead of an array of strings?
It seams more clean to call `this.running` than `this.prefixes[1]`.
We could alternatively use an object/Model...

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts:
```diff
@@ -6,180 +6,136 @@ import { Task } from '../models/task';
+    return `${this.prefixes[2]} ${this.suffix(metadata)}`;
+  }
+
+  constructor(prefixes: string[], suffix: (metadata) => string, errors: (metadata) => object) {
```
I think it would make more sense to replace `prefixes` with `states` and suffix with `ids`/`identification`/`unit`, since they are not always used in the beginning or end of the phrase.

This would also be reflected in the defaults variables names.

alfredodeza commented on src/pybind/mgr/dashboard/HACKING.rst:
```diff
@@ -1026,68 +1026,88 @@ updates its progress:
+  done. It consists of a list of three verbs for each different state f.e.
+  ``['create', 'Creating', 'Created']``.
+
+1. Failed messages always start with ``'Failed to '`` and should be continued
```
when enumerating, these can be taken care of by the doc system, so using `#` would automatically be rendered as the proper number

alfredodeza commented on src/pybind/mgr/dashboard/HACKING.rst:
```diff
@@ -1026,68 +1026,86 @@ updates its progress:
+``TaskManagerMessageService.messages``.  This has to be done to achieve
+consistency among all tasks and states.
+
+Operation Object
```
is this supposed to be a title?

Devp00l replied:
This functions like a `describe` in HTML see [here](https://github.com/ralsina/rst-cheatsheet/blob/master/rst-cheatsheet.rst#lists)

alfredodeza commented on src/pybind/mgr/dashboard/HACKING.rst:
```diff
@@ -1026,68 +1026,86 @@ updates its progress:
+consistency among all tasks and states.
+
+Operation Object
+  Ensures consistency among all tasks. It consists of three verbs for each
```
why is this indented?

Devp00l replied:
This is indented so it works like a `describe` in HTML.

alfredodeza commented on src/pybind/mgr/dashboard/HACKING.rst:
```diff
@@ -1026,68 +1026,86 @@ updates its progress:
+  different state f.e.
+  ``{running: 'Creating', failure: 'create', success: 'Created'}``.
+
+1. Put running operations in present participle f.e. ``'Updating'``.
```
the doc mechanism can take care of numbering this list using `#` instead of actual numbers

Devp00l commented:
@tspmelo and  @alfredodeza - addressed all comments

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-wrapper.service.ts:
```diff
@@ -51,7 +51,7 @@ export class TaskWrapperService {
       NotificationType.info,
       this.taskManagerMessageService.getRunningMessage(task),
-      this.taskManagerMessageService.getDescription(task)
+      this.taskManagerMessageService.getErrorTitle(task)
```
This can be remove.
We no longer need a message for this notification and atm it is showing an error message.

Devp00l replied:
We still need the title for consistency among all other tasks. This way every notification has a title, no mixing like before.

tspmelo replied:
The title is the line above the one I'm suggesting we remove.
The ErrorTitle is being passed as the message.

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/notification.service.ts:
```diff
@@ -75,9 +75,12 @@ export class NotificationService {
    * @returns The timeout ID that is set to be able to cancel the notification.
    */
-  show(type: NotificationType, message: string, title?: string, options?: any) {
+  show(type: NotificationType, title: string, message?: string, options?: any) {
```
Please update the documentation for this method.

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts:
```diff
@@ -5,204 +5,161 @@ import { FinishedTask } from '../models/finished-task';
+    return this.messages[task.name] || this.defaultMessage;
+  }
+
+  getSuccessMessage(task: FinishedTask) {
```
Since we are always using the value of this method as the title in the `show` method, I think this should be rename `getSuccessTitle`.

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-manager-message.service.ts:
```diff
@@ -5,204 +5,161 @@ import { FinishedTask } from '../models/finished-task';
+    return this._getTaskMessage(task).failure(task.metadata);
   }
 
   getRunningMessage(task: Task) {
```
Since we are always using the value of this method as the title in the `show` method, I think this should be rename `getRunningTitle`.

Devp00l commented:
Addressed all comments

tspmelo commented on src/pybind/mgr/dashboard/frontend/src/app/shared/services/task-wrapper.service.ts:
```diff
@@ -50,8 +50,8 @@ export class TaskWrapperService {
-      this.taskManagerMessageService.getRunningMessage(task),
-      this.taskManagerMessageService.getDescription(task)
+      this.taskMessageService.getRunningTitle(task),
+      this.taskMessageService.getErrorTitle(task)
```
This should be removed.
We don't have any description to show when the task is running.

Devp00l commented:
Addressed all comments

![cleaner-notifications](https://user-images.githubusercontent.com/16167865/44151200-21197690-a0a2-11e8-8961-06f88ae934f6.png)

Devp00l commented:
jenkins retest this please

Devp00l commented:
@alfredodeza I have addressed all your comments, please review this PR again for confirmation

Devp00l commented:
@alfredodeza could you approve this PR than so that it can be merged?
